source/MaterialXRuntime/Codegen/RtSubGraphImpl.cpp

```diff
@@ -0,0 +1,55 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Codegen/RtSubGraphImpl.h>
+#include <MaterialXRuntime/Tokens.h>
+
+#include <MaterialXRuntime/Private/PvtPath.h>
+#include <MaterialXRuntime/Private/PvtPrim.h>
+
+namespace MaterialX
+{
+
+DEFINE_TYPED_SCHEMA(RtSubGraphImpl, "nodeimpl:subgraphimpl");
+
+RtPrim RtSubGraphImpl::createPrim(const RtToken& typeName, const RtToken& name, RtPrim parent)
+{
+    if (typeName != _typeInfo.getShortTypeName())
+    {
+        throw ExceptionRuntimeError("Type names mismatch when creating prim '" + name.str() + "'");
+    }
+    if (!parent.getPath().isRoot())
+    {
+        throw ExceptionRuntimeError("A compoundimpl prim can only be created at the top / root level");
```

Comment:
Was this previously called compoundimpl? Should probably be renamed, no?

Reply:
Good catch, will fix it!